services/catalog/tests/unit/with_dbs/test_api_routes_services__inaccessible.py

```diff
@@ -0,0 +1,370 @@
+    response = client.post(
+        f"{url}",
+        headers={"x-simcore-products-name": target_product},
+        json={
+            "services_to_check": [
+                {
+                    "key": service_to_check["key"],
+                    "version": service_to_check["version"],
+                },
+            ],
+            "for_gid": 1,
+        },
+    )
+    assert response.status_code == 200
+    data = parse_obj_as(list[UserInaccessibleService], response.json())
+    assert user_id not in {item.user_id for item in data}
+    assert len(data) == 4
+
+    # We share the service with other user primary group
+    insert_user_id, insert_user_primary_gid = users_data[0]
+    async with sqlalchemy_async_engine.begin() as conn:
+        await conn.execute(
+            services_access_rights.insert()
+            .values(
+                key=service_to_check["key"],
+                version=service_to_check["version"],
+                gid=insert_user_primary_gid,
+                execute_access=True,
+                write_access=True,
+            )
+            .returning(services_access_rights)
+        )
+
+    url = URL("/v0/services:inaccessible")
+    response = client.post(
+        f"{url}",
+        headers={"x-simcore-products-name": target_product},
+        json={
+            "services_to_check": [
+                {
+                    "key": service_to_check["key"],
+                    "version": service_to_check["version"],
+                },
+            ],
+            "for_gid": 1,
+        },
+    )
+    assert response.status_code == 200
+    data = parse_obj_as(list[UserInaccessibleService], response.json())
+    response_users = {item.user_id for item in data}
+    assert user_id not in response_users
+    assert insert_user_id not in response_users
+    assert len(data) == 3
+
+
+async def test_inaccessible_services_for_standard_group(
+    mock_catalog_background_task: None,
+    director_mockup: MockRouter,
+    client: TestClient,
+    user_groups_ids: list[int],
+    user_id: UserID,
+    products_names: list[str],
+    service_catalog_faker: Callable,
+    services_db_tables_injector: Callable,
+    sqlalchemy_async_engine: AsyncEngine,
+    users_data: list[(UserID, GroupID)],
+):
+    _, _, _, team_yoda_gid = user_groups_ids
+    target_product = products_names[1]  # s4l
+    # create some fake services
+    NUM_SERVICES = 3
+    fake_services = [
+        service_catalog_faker(
+            f"simcore/services/dynamic/jupyterlab_{s}",
+            "1.0.1",
+            team_access=None,
+            everyone_access=None,
+            product=target_product,
+        )
+        for s in range(NUM_SERVICES)
+    ]
+    # injects fake data in db
+    await services_db_tables_injector(fake_services)
+
+    service_to_check = fake_services[0][0]  # --> service_meta_data table format
+
+    # There are no users in this standard group therefore there the output should be empty
```

Comment:
I am not sure I agree with this test actually.
As the user I want to share my project with that group.
the fact is that the services inside the project are not shared with that group right? so does it make sense that it returns 200, as in everything is ok?

use-case:
- someone creates that group
- as a user I want to check I can share my project with that group
- it will return 200 OK, no issues.
- I share the project
- later on users get assigned to that group
--> they cannot access the project because the service was actually never shared with that group.